manifests/init.pp

```diff
@@ -21,75 +21,133 @@
+    ensure   => installed,
+    provider => chocolatey,
+    source   => $_server_package_url,
   } ->
 
   # add windows features
   windowsfeature { 'Web-WebServer':
-  installmanagementtools => true,
+    installmanagementtools => true,
   } ->
   windowsfeature { 'Web-Asp-Net45':
   } ->
 
-  # remove default web site
-  iis::manage_site {'Default Web Site':
-    ensure        => absent,
-    site_path     => 'any',
-    app_pool      => 'DefaultAppPool'
-  } ->
+  # disable default web site
+  if $disable_default_website {
+    iis::manage_site {'Default Web Site':
+      ensure   => stopped,
+      app_pool => 'DefaultAppPool',
+    }
+  }
 
-  # application in iis
+  # create application pool in iis
   iis::manage_app_pool { "${_chocolatey_server_app_pool_name}":
     enable_32_bit           => true,
     managed_runtime_version => 'v4.0',
   } ->
+
+  # create website for chocolatey server
   iis::manage_site {'chocolatey.server':
-    site_path     => $_chocolatey_server_location,
-    port          => "${_chocolatey_server_app_port}",
-    ip_address    => '*',
-    app_pool      => "${_chocolatey_server_app_pool_name}",
-  } ->
+    site_path  => $_chocolatey_server_location,
+    port       => "${_chocolatey_server_app_port}",
+    ip_address => '*',
+    app_pool   => "${_chocolatey_server_app_pool_name}",
+  }
 
   # lock down web directory
   acl { "${_chocolatey_server_location}":
-    purge      => true,
-    inherit_parent_permissions  => false,
-    permissions => [
-     { identity => 'Administrators', rights => ['full'] },
-     { identity => 'IIS_IUSRS', rights => ['read'] },
-     { identity => 'IUSR', rights => ['read'] },
-     { identity => "IIS APPPOOL\\${_chocolatey_server_app_pool_name}", rights => ['read'] }
-   ],
-  } ->
-  acl { "${_chocolatey_server_location}/App_Data":
-    permissions => [
-     { identity => "IIS APPPOOL\\${_chocolatey_server_app_pool_name}", rights => ['modify'] },
-     { identity => 'IIS_IUSRS', rights => ['modify'] }
-   ],
+    purge                      => true,
+    inherit_parent_permissions => false,
+    permissions                => [
+      { identity => 'Administrators',
+      rights     => ['full'] },
+      { identity => 'IIS_IUSRS',
+      rights     => ['read'] },
+      { identity => 'IUSR',
+      rights     => ['read'] },
+      { identity => "IIS APPPOOL\\${_chocolatey_server_app_pool_name}",
+      rights     => ['read'] }
+    ],
+    require                    => Iis::manage_app_pool["${_chocolatey_server_app_pool_name}"],
+  }
+
+  # only set permissions if an alternate package folder is undefined
+  unless $packages_folder {
+    # ensure app_data folder is created
+    file { "${_chocolatey_server_location}/App_Data":
+      ensure => directory,
+    }
+
+    # ensure packages folder is created
+    file { "${_chocolatey_server_location}/App_Data/Packages":
+      ensure  => directory,
+      require => File["${_chocolatey_server_location}/App_Data"],
```

Comment:
For some reason, when I tried it on Windows 2012 R2 and set this to enabled, the apppool would stop shortly after starting. I can revisit and test to see if it still happens.  Would this be something that should be parametized?